### ISSUE TITLE
Run the agent without net.ipv4.conf.all.route_localnet?

### ISSUE THREAD

**Issue by felipeochoa** [aws/amazon-ecs-agent#997]
The approach recommended in [the install guide](http://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-agent-install.html) has the agent listening on `lo:51679` and using iptables to forward packets coming in on port 80 to the agent.

I would like to run the agent with `net.ipv4.conf.all.route_localnet` disabled to avoid having to write a bunch of other iptables rules to keep the local routing locked down. I think starting the agent with `-p 0.0.0.0:51679:51679` instead of `--net=host` would enable this setup, since the only NAT rule required then would be to map port 80 to port 51679, with `docker-proxy` handling the path to the container.

Are there other reasons for running the agent on the host net? I noticed that the original deployment recommendation was to expose `51678` to `127.0.0.1:51678:51678`, but couldn't figure out why that changed (around 58775e3).

**Comment by aaithal** [comment]
Hi @felipeochoa, thanks for reaching out to us regarding this. 

We decided against using the `docker0` bridge for serving credentials for a couple of reasons. [1] We found that under high load/contention, containers would actually fail to reach the ECS Agent on the bridge and [2] We did not want credentials going over the `docker0` bridge as any listeners on the bridge will be able to intercept these credentials.

Note that the ECS control plane and telemetry traffic would also share the `docker0` bridge with other containers and we didn't want that to cause more contention on the bridge as well.
